src/ComunicaEngine.js

```diff
@@ -22,23 +23,38 @@ export default class ComunicaEngine {
    * Creates an asynchronous iterable of results for the given SPARQL query.
    */
   async* execute(sparql, source) {
-    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql))
+    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql)) {
       yield* this.executeUpdate(sparql, source);
-
-    // Load the sources if passed, the default sources otherwise
-    const sources = await (source ? this.parseSources(source) : this._sources);
-    if (sources.length !== 0) {
-      // Execute the query and yield the results
-      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
-      yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+    }
+    else {
+      // Load the sources if passed, the default sources otherwise
+      const sources = await (source ? this.parseSources(source) : this._sources);
+      if (sources.length !== 0) {
+        // Execute the query and yield the results
+        const queryResult = await this._engine.query(sparql, { sources, ...this._options });
+        yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+      }
     }
   }
 
   /**
    * Creates an asynchronous iterable with the results of the SPARQL UPDATE query.
    */
   async* executeUpdate(sparql, source) {
-    throw new Error(`SPARQL UPDATE queries are unsupported, received: ${sparql}`);
+    // Load the sources if passed, the default sources otherwise
+    let sources = await (source ? this.parseSources(source) : this._destination);
+    if (sources.length !== 0) {
+      // Update queries can only handle a single source
+      if (sources.length > 1)
+        sources = [sources[0]];
+      // Execute the query and yield the results
+      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
+      if (queryResult.type !== 'update')
+        throw new Error(`Update query returned unexpected result type: ${queryResult.type}`);
```

Comment:
That wouldn't happen, would it?

Reply:
This is the pattern that I tend to follow when using Comunica in Typescript projects due to the fact that the engine returns a `QueryResult`, doing this check confirms that it is indeed of type `UpdateResult` allowing you to access the `updateResult` property without getting type errors.

You can see I add a similar check [here](https://github.com/LDflex/LDflex-Comunica/blob/a00c0b24ee05d263a483efa7eeebe3a0f460b17c/src/ComunicaEngine.ts#L62) for the bindings result in the PR to add Typescript.

I think this should be kept (or at least added back in at the Typescriptification PR) until Comunica does something like having a type overload for the query method as briefly discussed in [here](https://github.com/comunica/comunica/issues/907)